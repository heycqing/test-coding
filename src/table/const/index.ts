export const ORDER = {
  asce: 'asce',
  desc: 'desc',
}

export const paginationPageIndex = {
  pre: 'pre',
  next: 'next',
  first: 'first',
  last: 'last',
}
