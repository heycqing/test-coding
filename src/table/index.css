/* .columns-desc{
    display: inline-flex;
    margin: 0 10px;
    color: #666;
    flex-direction: column;
}

.pagination > span{
    margin: 0 6px;
} */