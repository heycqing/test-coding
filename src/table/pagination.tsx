/**
 * Created by uedc on 2021/10/11.
 */

 import { defineComponent } from '@vue/composition-api'
 // import { tableProps } from './types'
 
 export default defineComponent({
     name: 'Pagination',
     props: {
         size: { type: String, required: true },
     },
     setup(props, { slots }) {
         return () => {
             return (
                 <div class="pagination">
                    <span>首页</span>
                    <span onClick={props.onGetPrepage}>上一页</span>
                    <span onClick={props.onGetNextpage}>下一页</span>
                    <span>尾页</span>
                 </div>
             )
         }
     },
 })