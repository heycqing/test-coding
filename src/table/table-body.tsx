/**
 * Created by uedc on 2021/10/11.
 */

import { defineComponent } from '@vue/composition-api'
// import { tableProps } from './types'

export default defineComponent({
    name: 'TableBody',
    props: {
        dataSource: { type: Array, required: true },
        columns: { type: Array, required: true }
    },
    setup(props, { slots }) {
        return () => {
            return (
                <tbody>
                    {/* 生成数据 */}
                    {props.dataSource.map((item) => {
                            return (
                                <tr>
                                    {/* 依据columns生成 */}
                                    {props.columns.map(keyItem => keyItem.key).map(colItem => {
                                        return <td>{item[colItem]}</td>
                                    })}
                                </tr>
                            )
                        })}
                </tbody>
            )
        }
    },
})