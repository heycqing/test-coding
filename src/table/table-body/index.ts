/**
 * Created by uedc on 2021/10/11.
 */

import TableBody from './table-body'
export { TableBody }

