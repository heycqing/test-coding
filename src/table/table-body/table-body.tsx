/**
 * Created by uedc on 2021/10/11.
 */

import { defineComponent } from '@vue/composition-api'

export default defineComponent({
    name: 'TableBody',
    props: {
        dataSource: { type: Array, required: true },
        columns: { type: Array, required: true }
    },
    setup(props, { slots }) {
        return () => {
            return (
                <tbody class="table-body">
                    {/* 生成数据 */}
                    {props.dataSource.map((item) => {
                            return (
                                <div>
                                    {slots.rowContent ? slots.rowContent : (
                                        <th>
                                        {props.columns.map(keyItem => keyItem.key).map(colItem => {
                                            return <td>{item[colItem]}</td>
                                        })}
                                        </th>
                                    )}
                                </div>
                            )
                        })}
                </tbody>
            )
        }
    },
})