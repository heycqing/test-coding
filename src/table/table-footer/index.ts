/**
 * Created by uedc on 2021/10/11.
 */

import Pagination from './pagination'
export { Pagination }


