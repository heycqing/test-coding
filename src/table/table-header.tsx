/**
 * Created by uedc on 2021/10/11.
 */

import { defineComponent } from '@vue/composition-api'

const ORDER = {
    asce: 'asce',
    desc: 'desc'
}

export default defineComponent({
    name: 'TableHeader',
    props: {
        columns: { type: Array, required: true },
        dataSource: {type: Array, required: true},
        handleSetSourceData: {type: Function, required: true}
    },
    setup(props) {
        return () => {
            return (
                <thead>
                    <tr>
                        {props.columns.map(item => {
                                return (
                                <th>
                                    {item.title}
                                    {/* 排序 */}
                                    {item.order && 
                                        (<span class="columns-desc">
                                            <span onClick={() => props.handleSetSourceData(props.dataSource, ORDER.asce, item.key)}>升序</span>
                                            <span onClick={() => props.handleSetSourceData(props.dataSource, ORDER.desc, item.key)}>降序</span>
                                        </span>)}
                                </th>)
                        })}
                    </tr>
                </thead>
            )
        }
    },
})

