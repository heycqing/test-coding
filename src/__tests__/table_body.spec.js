import { mount } from '@vue/test-utils'
import { TableBody } from '../table/table-body'
import { tableArr, tableColumns } from '../test-mock/mockdata.test'

describe('TableBody', () => {

  const TableMount = options => mount(TableBody, options)

  test('render', () => {
    const wrapper = TableMount({
      propsData: {
        dataSource: tableArr,
        columns: tableColumns,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => {
      wrapper.vm.$forceUpdate()
      wrapper.vm.$destroy()
    })
  })
})
