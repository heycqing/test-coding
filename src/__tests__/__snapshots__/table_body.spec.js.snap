// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody render 1`] = `
"<tbody>
  <tr>
    <td>小明</td>
    <td>man</td>
    <td>16</td>
  </tr>
  <tr>
    <td>小红</td>
    <td>woman</td>
    <td>12</td>
  </tr>
  <tr>
    <td>小明</td>
    <td>woman</td>
    <td>13</td>
  </tr>
  <tr>
    <td>小明</td>
    <td>woman</td>
    <td>14</td>
  </tr>
  <tr>
    <td>小明</td>
    <td>man</td>
    <td>25</td>
  </tr>
  <tr>
    <td>小刚</td>
    <td>man</td>
    <td>6</td>
  </tr>
  <tr>
    <td>小明</td>
    <td>man</td>
    <td>7</td>
  </tr>
</tbody>"
`;
